indexing-service/src/main/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTask.java

```diff
@@ -77,7 +83,12 @@ public KillUnusedSegmentsTask(
         interval,
         context
     );
+    if (null != maxSegmentsToKill) {
+      Preconditions.checkArgument(maxSegmentsToKill > 0, "maxSegmentsToKill must be > 0");
+    }
```

Comment:
Since this is user-facing, we could use `InvalidInput.exception()`:
```suggestion
    if (null != maxSegmentsToKill && maxSegmentsToKill <= 0) {
       throw InvalidInput.exception("maxSegmentsToKill [%d] is invalid. It must be a positive integer.", maxSegmentsToKill);
    }
```